pvnet_app/data.py

```diff
@@ -32,143 +32,197 @@ def download_all_sat_data():
+        fill_times = pd.date_range(t0 + timedelta(minutes=(-delay_mins + 5)), t0, freq="5min")
+
+        ds_sat = ds_sat.reindex(time=np.concatenate([ds_sat.time, fill_times]), fill_value=np.nan)
 
         # Re-save inplace
         os.system(f"rm -rf {sat_path}")
         ds_sat.to_zarr(sat_path)
-    
+
     return delay_mins
-        
-    
-def check_model_inputs_available(data_config_filename, sat_delay_mins):
-    """Checks whether the model can be run given the current satellite delay"""
-    
+
+
+def check_model_inputs_available(
+    data_config_filename, all_satellite_datetimes, t0, data_freq_minutes
+):
+    """Checks whether the model can be run given the current satellite delay
+
+    Args:
+        data_config_filename: Path to the data configuration file
+        all_satellite_datetimes: All the satellite datetimes available
+        t0: The time the model is trying to forecast
+        data_freq_minutes: The frequency of the satellite data. This can be 5 or 15 minutes.
+
+    """
+
     data_config = load_yaml_configuration(data_config_filename)
-    
+
     available = True
-    
-    # check satellite if using
+
+    # check satellite if using
     if hasattr(data_config.input_data, "satellite"):
         if data_config.input_data.satellite is not None:
 
-            # Take into account how recently the model tries to slice satellite data from
+            # Take into account how recently the model tries to slice satellite data from
             max_sat_delay_allowed_mins = data_config.input_data.satellite.live_delay_minutes
 
             # Take into account the dropout the model was trained with, if any
-            if data_config.input_data.satellite.dropout_fraction>0:
+            if data_config.input_data.satellite.dropout_fraction > 0:
                 max_sat_delay_allowed_mins = max(
                     max_sat_delay_allowed_mins,
-                    np.abs(data_config.input_data.satellite.dropout_timedeltas_minutes).max()
+                    np.abs(data_config.input_data.satellite.dropout_timedeltas_minutes).max(),
+                )
+
+            # get start and end satellite times
+            history_minutes = data_config.input_data.satellite.history_minutes
+
+            # we only check every 15 minutes, as ocf_datapipes resample from 15 to 5 if necessary.
+            # We need to include the end point also
+            freq = f"{data_freq_minutes}min"
+            logger.info(
+                f"Checking satellite data for {t0=} with history {history_minutes=} "
+                f"and freq {freq=}, for {max_sat_delay_allowed_mins=}"
+            )
+            expected_datetimes = pd.date_range(
+                t0 - timedelta(minutes=int(history_minutes)),
+                t0
+                - timedelta(minutes=int(max_sat_delay_allowed_mins))
+                + timedelta(minutes=int(data_freq_minutes)),
```

Comment:
I think we shouldn't add `data_freq_minutes` here?  `pd.date_range(t_start, t_end)` is inclusive of the endpoints `t_start` and `t_end`. So if `max_sat_delay_allowed_mins=0` and the model was trained with zero satellite delay and no dropout we need time `t0`. The way the code is now would include time `t0+ timedelta(minutes=int(data_freq_minutes))` in the expected datetimes, which we don't need?

Reply:
Ah my mistake. I assuem it wasnt inclusive of the last timestamp, but Ive just checked it and it does. 

I.e.
```
 pd.date_range('2021-01-01','2021-01-02',freq='1h')
```
does include `2021-01-02`

ive removed it in the commit